dvc/rwlock.py

```diff
@@ -176,3 +190,62 @@ def rwlock(tmp_dir, fs, cmd, read, write, hardlink):
         with _edit_rwlock(tmp_dir, fs, hardlink) as lock:
             _release_write(lock, info, wchanges)
             _release_read(lock, info, rchanges)
+
+
+def check_rwlock(
+    tmp_dir: str,
+    fs: "FileSystem",
+    hardlink: bool = False,
+    autocorrect: bool = False,
+) -> bool:
```

Comment:
Since this function uses exp executor pidfiles and PIDs to determine whether or not a lock is valid, it probably doesn't belong in `dvc/rwlock.py`. It's specific to experiment execution, and is not a general lockfile method.

If you think this is functionality we could re-use for other locks in DVC, one option would be to implement it as a generic method in `dvc/rwlock.py` that takes a callback function which returns True/False depending on whether or not the lock is valid. And then when we use it in the exp methods we would pass a callback function that checks the pidfile.